src/modules/BootStorage.rb

```diff
@@ -241,26 +235,23 @@ def prep_partitions
     end
 
     def disk_with_boot_partition
-      boot_device = BootPartitionDevice()
-
-      partition = Storage::Partition.find_by_name(staging, boot_device)
-      partitionable = partition.partition_table.partitionable
+      partitionable = boot_partition.partition_table.partitionable
```

Comment:
This can be directly `boot_partition.partitionable` or even `boot_partition.disk`.